### PR TITLE
Change our use of Functor.map (via <$>) to a more targeted use of ..

### PR DESCRIPTION
**Pull request by tobiasgrosser**
RegionSignature.map and Signature.map. This makes proof states easier to read, and allows for unfolding one of the instances of Functor.map at a time.

This is pulled out of https://github.com/opencompl/ssa/pull/225 as an independent useful change.

Co-authored-by: Siddharth Bhat <siddu.druid@gmail.com>

### CODE REVIEW

**Comment by tobiasgrosser** [comment]
@bollu authored this change, so we should get a review from @alexkeizer or @goens.

**Review by goens — CHANGES_REQUESTED** [review]
The code lgtm, just a bit worried how this will interact with #191 (I guess we'll also merge this with that PR if merged?)

I have one comment about the metadata though: If @bollu authored this change, maybe he should be in the commits? At least with a co-authored-by tag (see https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors).

**Pull request merged**
